IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetSsoCookiesResponse.m

```diff
@@ -61,20 +61,34 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)er
         }
         
         NSString *ssoCookiesString = json[@"sso_cookies"];
-        
         if ([NSString msidIsStringNilOrBlank:ssoCookiesString])
         {
-            self.prtHeaders = nil;
-            self.deviceHeaders = nil;
-            self.success = YES;
             return self;
         }
         
         NSDictionary *ssoCookiesJson = [ssoCookiesString msidJson];
+        NSError *convertPrtHeaderError = nil;
+        NSError *convertDeviceHeaderError = nil;
+        _prtHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_PRT_HEADERS error:&convertPrtHeaderError];
+        _deviceHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_DEVICE_HEADERS error:&convertDeviceHeaderError];
         
-        _prtHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_PRT_HEADERS error:error];
-        
-        _deviceHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_DEVICE_HEADERS error:error];
+        if(convertPrtHeaderError && convertDeviceHeaderError)
+        {
+            if(error)
+            {
+                *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, [NSString stringWithFormat:@"Cannot convert credential headers from json response. PrtHeader error is %@. Device header error is %@", convertPrtHeaderError.description, convertDeviceHeaderError.description], nil, nil, nil, nil, nil, YES);
+            }
+            
+            self.success = NO;
```

Comment:
In this error case, do you want to return nil self or not?

Reply:
In this case, it will return a valid response, which means it contains an error and a BOOL as not successful.